sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryStrategy.java

```diff
@@ -57,6 +57,6 @@ default boolean shouldRetry(HttpResponse httpResponse) {
      * @return Whether a retry should be attempted.
      */
     default boolean shouldRetryException(Throwable throwable) {
-        return true;
+        return throwable instanceof Exception;
```

Comment:
wow.

Reply:
btw. .NET is more specific here see https://github.com/Azure/azure-sdk-for-net/blob/4b61e879b3b1f620a0a5efd9f14b3f4579e7e1c4/sdk/core/Azure.Core/src/ResponseClassifier.cs#L41-L42 .

I wonder if we should be more specific as well. for example if user injects retry level policy that has a bug that throws NPE we shouldn't retry that.